### PR TITLE
fix(headless): clear response/results when search contains an error

### PR DESCRIPTION
**Pull request by ThibodeauJF**
Moved from the controller's level to the slice's level
https://coveord.atlassian.net/browse/KIT-833

### CODE REVIEW

File: packages/headless/src/features/search/search-slice.ts [hunk 1]
```diff
@@ -8,7 +8,13 @@ function handleRejectedSearch(
   state: SearchState,
   action: ReturnType<SearchAction['rejected']>
 ) {
-  state.error = action.payload ? action.payload : null;
+  const hasError = action.payload ? action.payload : null;
```
Comment (ThibodeauJF) [inline packages/headless/src/features/search/search-slice.ts:11]: I don't think a switch case is necessary, we get null when we abort the request

Comment (ThibodeauJF) [reply packages/headless/src/features/search/search-slice.ts:11]: I'll check for the accessToken renewal, that could be a case where we dont want to clear the results

**Review by btaillon — APPROVED** [review]

File: packages/headless/src/features/search/search-slice.ts (hunk 1 above)
Comment (ThibodeauJF) [reply packages/headless/src/features/search/search-slice.ts:11]: Tested and works well for Token renewal, the hasError remains false

**Review by nbuissoncoveo — APPROVED** [review]

**Review by olamothe — APPROVED** [review]
What should happen to all facet sets ? 

My first intuition would be that the same logic should apply to them as well: there's no "facet results" in the response when you get an error from the API, so we clear them from the state/reset to empty.

At the same time, you probably don't want to clear the facet values that are currently active (selected), since that's still technically "true" for the state that they are selected, just that the count is "unknown".

**Comment by ThibodeauJF** [comment]
It is the case that it clears the whole response: `state.response = getSearchInitialState().response;`
Results are in 2 places, so we clear it there too `state.results = [];`
The tests also cover this

File: packages/headless/src/features/search/search-slice.ts (hunk 1 above)
Comment (samisayegh) [reply packages/headless/src/features/search/search-slice.ts:11]: Would `error` be a better name for the variable?

```
const error = action.payload || null;
```

**Review by samisayegh — APPROVED** [review]

**Pull request merged**